test/1_stdlib/NewString.swift

```diff
@@ -114,6 +114,10 @@ func nonASCII() {
   var nsSliceUTF16 = newNSUTF16[i2..<i8] as NSString
   print("  \(repr(nsSliceUTF16))")
 
+  // CHECK-NEXT: fast C-String contents: nil
+  let cntgStr = unsafeBitCast(nsSliceUTF16, to: _NSContiguousString.self)
```

Comment:
@jrose-apple thanks for keeping up the fight against `unsafeBitCast` of pointers and references. `unsafeDownCast` works as long as the source is statically AnyObject and dynamically the destination class, which looks like the case.

`_unsafeReferenceCast` handles a broader set of class-related casts including optional references and class existentials. It also does away with the debugPrecondition (though we should still have a sanity check if the type system can handle all the cases now).